devito/propagator.py

```diff
@@ -4,20 +4,48 @@
 from sympy import symbols, IndexedBase, Indexed
 from function_manager import FunctionDescriptor
 from collections import Iterable
+from at_controller import get_at_block_size, get_optimal_block_size
 
 
 class Propagator(object):
-    def __init__(self, name, nt, shape, spc_border=0, forward=True, time_order=0, openmp=False, profile=False, cache_blocking=False, block_size=5):
+    def __init__(self, name, nt, shape, spc_border=0, forward=True, time_order=0, openmp=False, profile=False,
+                 cache_blocking=False, block_size=None, auto_tune=False):
         self.t = symbols("t")
+
         self.cache_blocking = cache_blocking
-        if(isinstance(block_size, Iterable)):
-            if(len(block_size) == len(shape)):
-                self.block_sizes = block_size
+        self.auto_tune = auto_tune
+
+        self.spc_order = spc_border * 2  # setting space order
+        self.time_order = time_order
+
+        if self.cache_blocking:
```

Comment:
I think the number of flags here is getting confusing. I propose the following: 
- Remove `auto-tune` as a flag
- Default value of  `block_size` is `None`. Default value of `cache_blocking` is `False`.
- If `block_size` is not `None` and `cache_blocking` is `False`, show an error to the user since they are probably expecting a behaviour that is being overriden.
- If `cache_blocking` is `True` and `block_size` is `None`, use auto-tuning. Show a message to the user informing them about this choice.
- If `cache_blocking` is `True` and `block_size` is provided, use it.